pkg/client/kubelet.go

```diff
@@ -135,7 +136,43 @@ func (c *HTTPKubeletClient) GetPodStatus(host, podNamespace, podID string) (api.
 }
 
 func (c *HTTPKubeletClient) HealthCheck(host string) (probe.Result, error) {
-	return httprobe.DoHTTPProbe(fmt.Sprintf("%s/healthz", c.url(host)), c.Client)
+	url := fmt.Sprintf("%s/healthz", c.url(host))
+	res, err := c.Client.Get(url)
+	if err != nil {
+		glog.V(1).Infof("HTTP probe error: %v", err)
+		return probe.Failure, nil
+	}
+
+	defer res.Body.Close()
+	response, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		glog.V(1).Infof("HTTP ready body error: %v", err)
+		return probe.Failure, nil
+	}
+
+	type HealthResponse struct {
+		Hostname string
+		Health   string
+	}
+	var healthResponse HealthResponse
```

Comment:
If this approach is okay this struct can be generalized and moved somewhere where it can be shared by kubelet and master code. I think one of the types.go files from what I have read sofar